ddtrace/internal/telemetry/writer.py

```diff
@@ -621,14 +622,6 @@ def _fork_writer(self):
         # Avoid sending duplicate events.
         # Queued events should be sent in the main process.
         self.reset_queues()
```

Comment:
```suggestion
        self.reset_queues()
        if self.status == ServiceStatus.STOPPED:
           return

       if self._is_periodic:
           self.stop(join=False)

        # Enable writer service in child process to avoid interpreter shutdown
        # error in Python 3.12
        self.enable()
```